components/IndoorDirections/IndoorScenario.js

```diff
@@ -0,0 +1,70 @@
+import React, { useEffect } from 'react';
+import { G } from 'react-native-svg';
+import { HallXCoordinates } from '../../constants/HallXCoordinates';
+import { ClassGraph } from '../../constants/ClassGraph';
+import { DifferentFloorDirections } from './TypesOfDirections/DifferentFloorDirections';
+import { SameFloorDirections } from './TypesOfDirections/SameFloorDirections';
+import { DifferentBuildingDirections } from './TypesOfDirections/DifferentBuildingDirections';
+
+export function IndoorScenario(props) {
+    const rooms = HallXCoordinates();
+
+    if (props.from && props.to) {
+        const routeToTake = whichPathToTake(props.from.toString(), props.to.toString());
+        switch (routeToTake) {
+            case "SAME_FLOOR":
+                return <SameFloorDirections rooms={rooms} floor={props.floor} from={props.from.toString()} to={props.to.toString()}/>
+                break;
+            case "DIFFERENT_FLOOR":
+                return <DifferentFloorDirections rooms={rooms} floor={props.floor} from={props.from.toString()} to={props.to.toString()}/>;
+                break;
+            case "DIFFERENT_BUILDING":
+                return <DifferentBuildingDirections rooms={rooms} floor={props.floor} from={props.from.toString()} to={props.to.toString()}/>;
+                break;
+        }
+    }
+
+    
+    return(
+        <G></G>
+    );
+}
+
+/**
+ * Takes two classrooms & determines if the user needs to leave the
+ * building, switch floors or stay in the same floor.
+ * @param {*} from 
+ * @param {*} to 
+ */
+export const whichPathToTake = (from, to) => {
```

Comment:
is it used elsewhere? If no, remove the export. If yes, you should maybe create its own .js file for readability and usability purposes.

Reply:
It's being exported so that it can be tested. Otherwise, I can put it in another file if you want, but I found it useful to have it below where it's being used

Reply:
Sounds fair to me to have it there then. However, it means we should check if we test (unit) all stateless functions as well.